src/shared/script/shared-code/mathUtils.ts

```diff
@@ -48,6 +26,52 @@ export function divisors(num: PLNumber): PLVector<PLNumber> {
   return plVector(...divisors.map(plNumber))
 }
 
+// https://www.thepolyglotdeveloper.com/2015/04/determine-if-a-number-is-prime-using-javascript/
+export function isPrime(n: PLNumber): PLBool {
+  assertInteger(n.value)
+  if (n.value < 1) {
+    throw new Error('Number out of range (< 1)')
+  }
+  for (let i = 2; i < n.value; i++) {
+    if (n.value % i === 0) {
+      return plBool(false)
+    }
+  }
+  return plBool(n.value > 1)
+}
+
+// https://stackoverflow.com/questions/9083037/convert-a-number-into-a-roman-numeral-in-javascript#32851198
+const numToRoman = (num: number, result = ''): string => {
+  if (num < 1 && result === '') {
+    throw new Error('Invalid range (< 1)')
+  } else if (num > 3999 && result === '') {
+    throw new Error('Invalid range (> 3999)')
+  }
+  const map = {
+    M: 1000,
+    CM: 900,
+    D: 500,
+    CD: 400,
+    C: 100,
+    XC: 90,
+    L: 50,
+    XL: 40,
+    X: 10,
+    IX: 9,
+    V: 5,
+    IV: 4,
+    I: 1,
+  }
+  for (const key in map) {
+    if (num >= map[key]) {
+      if (num !== 0) {
+        return numToRoman(num - map[key], result + key)
```

Comment:
You do not need the `result` parameter if you do this
```suggestion
        return numToRoman(num - map[key]) + key
```